presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -437,6 +437,16 @@ public LookupSourceSupplier createLookupSourceSupplier(
         return createLookupSourceSupplier(session, joinChannels, hashChannel, filterFunctionFactory, sortChannel, searchFunctionFactories, Optional.empty());
     }
 
+    public PagesSpatialIndex createPagesSpatialIndex(
```

Comment:
It would be nice if it would be pluggable.

Reply:
Are you referring to having ability for plugins to provide custom optimizer rules and operators in addition to types and functions? If so, yes, that would be very nice. I hope we'll get there someday.